File: MyAnimeList/src/main/java/com/example/myanimelist/di/components/ReviewsRepositoryDI.java
package com.example.myanimelist.di.components;

import com.example.myanimelist.controllers.ReviewsController;
import com.example.myanimelist.di.modules.AnimeRepositoryModule;
import com.example.myanimelist.di.modules.DataBaseManagerModule;
import com.example.myanimelist.di.modules.UsersRepositoryModule;
import com.example.myanimelist.repositories.reviews.ReviewsRepository;
import dagger.Component;

import javax.inject.Singleton;

@Singleton
@Component(modules = {DataBaseManagerModule.class, AnimeRepositoryModule.class, UsersRepositoryModule.class})
public interface ReviewsRepositoryDI {
    ReviewsRepository build();
    void inject(ReviewsController controller);
}

File: MyAnimeList/src/main/kotlin/com/example/myanimelist/di/components/ReviewsRepositoryDI.kt
package com.example.myanimelist.di.components

import com.example.myanimelist.controllers.ReviewsController
import com.example.myanimelist.di.modules.DataBaseManagerModule
import com.example.myanimelist.repositories.reviews.ReviewsRepository
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [
    DataBaseManagerModule::class
, ])
interface ReviewsRepositoryDI {
    fun build() : ReviewsRepository
    fun inject(controller : ReviewsController)
}